src/concrete/ml/torch/hybrid_model.py

```diff
@@ -1,16 +1,44 @@
 """Implement the conversion of a torch model to a hybrid fhe/torch inference."""
 
+import ast
+import enum
+import io
+import sys
+import time
 import uuid
 from pathlib import Path
-from typing import List, Optional, Union
+from typing import Dict, List, Optional, Tuple, Union
 
+import numpy
+import requests
 import torch
 from concrete.fhe import Configuration
 from torch import nn
 from transformers import Conv1D
 
 from ..deployment.fhe_client_server import FHEModelClient, FHEModelDev
-from .compile import compile_torch_model
+from .compile import QuantizedModule, compile_torch_model
+
+
+class FHEMode(enum.Enum):
+    """Simple enum for different modes of execution of HybridModel."""
+
+    DISABLE = "disable"  # Use torch weights
+    REMOTE = "remote"  # Use remote FHE server
+    SIMULATE = "simulate"  # Use FHE simulation
+    CALIBRATE = "calibrate"  # Use calibration (to run before FHE compilation)
+
+
+def tuple_to_underscore_str(tup: Tuple) -> str:
+    """Convert a tuple to a string representation.
+
+    Args:
+        tup (Tuple): a tuple to change into string representation
+
+    Returns:
+        str: a string representing the tuple
+    """
+    return repr(tup).replace("(", "p_").replace(")", "_p").replace(", ", "_")
```

Comment:
why is this needed ? why replacing parenthesis with p_ or _p ?

Reply:
Makes it easier to parse.